core/advancedpolls.php

```diff
@@ -346,6 +352,23 @@ public function do_poll_modification($topic_data)
 	}
 
 	/**
+	 * Add the possible options to the template
+	 *
+	 * @param array	$post_data		The array of post data
+	 * @return void
+	 */
+	public function do_poll_block_modification($poll_options_template_data)
+	{
+		usort($poll_options_template_data, array($this, "order_by_votes"));
+		return $poll_options_template_data;
+	}
+
+	protected function order_by_votes($a, $b)
+	{
+		return (((int) $a['POLL_OPTION_RESULT'] < (int) $b['POLL_OPTION_RESULT']) ? 1 : (((int) $a['POLL_OPTION_RESULT'] > (int) $b['POLL_OPTION_RESULT']) ? -1 : (((int) $a['POLL_OPTION_ID'] > (int) $b['POLL_OPTION_ID']) ? 1 : -1))); 
```

Comment:
You can do that of course, it is understandable.
It wouldn't make any difference though, internally ternary operator will be executed the same way like an if/else statement :P